src/backend/web/renderers/team_renderer.py

```diff
@@ -182,8 +183,55 @@ def render_team_details(
                     None,
                 )
 
+            alliance_size = KNOWN_EVENT_TYPE_ALLIANCE_SIZES.get(
+                event.event_type_enum, 99
+            )
+
+            alliance = None
+            alliance_pick = None
+            if event.details and event.details.alliance_selections:
```

Comment:
It'd be nice if this could get broken out into a helper function (e.g. in a new file like `alliance_helper.py`) since there's some quite involved logic here that would be nice to unit test separately.